src/cider/nrepl/middleware/profile.clj

```diff
@@ -0,0 +1,134 @@
+                              :status :done
+                              :err table))
+        (t/send transport
+                (response-for msg
+                              :status :done
+                              :err (format "No profile data for %s." v))))
+      (t/send transport
+              (response-for msg
+                            :status :done
+                            :value (format "Var %s/%s is not bound." ns sym))))
+    (catch Exception e (prn :e e) (send-exception e msg transport))))
+
+(defn profile-summary
+  [{:keys [transport] :as msg}]
+  (try
+    (t/send transport
+            (response-for msg
+                          :status :done
+                          :err (with-out-str
+                                 (binding [*err* *out*] (p/print-summary)))))
+    (catch Exception e (send-exception e msg transport))))
+
+(defn clear-profile
+  [{:keys [transport] :as msg}]
+  (try
+    (p/clear-profile-data)
+    (t/send transport
+            (response-for msg
+                          :status :done
+                          :value "cleared"))
+    (catch Exception e (send-exception e msg transport))))
+
+(defn toggle-profile-ns
+  [{:keys [ns transport] :as msg}]
+  (try (let [profiled? (p/toggle-profile-ns (symbol ns))]
+         (t/send transport
+                 (response-for
+                  msg
+                  :status :done
+                  :value (if profiled? "profiled" "unprofiled"))))
+       (catch Exception e (send-exception e msg transport))))
+
+(defn is-var-profiled
+  [{:keys [ns sym transport] :as msg}]
+  (try (let [var (ns-resolve (symbol ns) (symbol sym))
+             profiled? (p/profiled? @var)]
+         (t/send transport
+                 (response-for
+                  msg
+                  :status :done
+                  :value (if profiled? "profiled" "unprofiled"))))
+       (catch Exception e (send-exception e msg transport))))
+
+(defn get-max-samples
+  [{:keys [transport] :as msg}]
+  (try (t/send transport
+               (response-for
+                msg
+                :status :done
+                :value (str (p/max-sample-count))))
+       (catch Exception e (send-exception e msg transport))))
+
+(defn normalize-max-samples [n]
+  (cond (and (sequential? n) (empty? n)) nil
+        (string? n) (Long/parseLong n)
+        :else n))
+
+(defn set-max-samples
+  [{:keys [max-samples transport] :as msg}]
+  (try (let [max-samples (normalize-max-samples max-samples)]
+         (p/set-max-sample-count max-samples)
+         (t/send transport
+                 (response-for
+                  msg
+                  :status :done
+                  :value (str (p/max-sample-count)))))
+       (catch Exception e  (send-exception e msg transport))))
+
+(defn handle-profile
+  [handler msg]
+  (let [{:keys [op] } msg]
+    (case op
+      "toggle-profile"
+      (toggle-profile msg)
+      "toggle-profile-ns"
+      (toggle-profile-ns msg)
```

Comment:
Great work @jmayaalv!

One minor comment; should the formatting of the case expression be this instead: 
```
(case op
  "toggle-profile" (toggle-profile msg)
  "..."  ...)
```

Would be a bit nicer to read I think.